src/components/ContributeDetails.js

```diff
@@ -0,0 +1,111 @@
+import PropTypes from 'prop-types';
+import { compose, withHandlers, withState } from 'recompose';
+import { format, addYears, addMonths } from 'date-fns/esm';
+import { pick } from 'lodash';
+
+import Container from './Container';
+import { Flex } from '@rebass/grid';
+import StyledButtonSet from './StyledButtonSet';
+import StyledInputField from './StyledInputField';
+import StyledInput from './StyledInput';
+import StyledSelect from './StyledSelect';
+import { P, Span } from './Text';
+import Currency from './Currency';
+
+const frequencyOptions = {
+  year: 'Yearly',
+  month: 'Monthly',
+};
+
+const addFrequency = {
+  year: addYears,
+  month: addMonths,
+};
+
+const enhance = compose(
+  withState('state', 'setState', ({ amountOptions, showFrequency }) => ({
+    amount: amountOptions[0] / 100,
+    totalAmount: amountOptions[0],
+    interval: showFrequency ? Object.keys(frequencyOptions)[0] : undefined,
+  })),
+  withHandlers({
+    onChange: ({ state, setState, onChange }) => newState => {
+      newState = { ...state, ...newState };
+      setState(newState);
+      onChange(pick(newState, ['totalAmount', 'interval']));
+    },
+  }),
+);
+
+const ContributeDetails = enhance(({ amountOptions, currency, showFrequency, onChange, state }) => (
+  <Container as="fieldset" border="none">
+    <Flex>
+      <StyledInputField label={`Amount (${currency})`} htmlFor="totalAmount">
+        {fieldProps => (
+          <StyledButtonSet
+            {...fieldProps}
+            combo
+            items={amountOptions}
+            selected={state.totalAmount}
+            onChange={totalAmount => onChange({ totalAmount, amount: totalAmount / 100 })}
+          >
+            {({ item }) => <Currency value={item} currency={currency} />}
+          </StyledButtonSet>
+        )}
+      </StyledInputField>
+      <Container maxWidth={100}>
+        <StyledInputField label="Other" htmlFor="totalAmount">
+          {fieldProps => (
+            <StyledInput
+              type="text"
```

Comment:
Having a text input here is not appropriate:
![selection_992](https://user-images.githubusercontent.com/1556356/50759495-724a9c80-1265-11e9-8a6f-d9361fb4b905.png)

Ideally we should use the [StyledInputAmount](https://opencollective-styleguide-uhsgdawcum.now.sh/#styledinputamount) for this

Reply:
I will fix the text input to only allow numbers and decimals, but the StyledInputAmount does not match the design for this component.